File: 240422/삽입 정렬 구현/implement-insertion-sort.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] arr = new int[n];
        for(int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }

        for(int i = 1; i < n; i++){
            int j = i - 1; // sorted part index
            int key = arr[i]; // 앞과 비교해서 삽입할 값

            while(j >= 0 && arr[j] > key){
                arr[j+1] = arr[j];
                j--;
            }
            arr[j + 1] = key;
        }

        for(int item: arr){
            System.out.print(item + " ");
        }
    }
}

File: 240422/삽입 정렬 구현/implement-insertion-sort.kt
import java.util.*

fun main() {
    val sc = Scanner(System.`in`)
    val n = sc.nextInt()
    val arr = IntArray(n){ sc.nextInt() }

    for(i in 1 until n){
        var j = i - 1 // sortedPartIdx
        var key = arr[i]
        while(j >= 0 && key < arr[j]){
            arr[j + 1] = arr[j]
            j--
        }
        arr[j + 1] = key
    }
    
    for(item in arr){
        print("${item} ")
    }
}